src/common/ServerCompatBanner.js

```diff
@@ -0,0 +1,101 @@
+/* @flow strict-local */
+
+import React from 'react';
+import { View } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import store from '../boot/store';
+import { createStyleSheet, HALF_COLOR } from '../styles';
+import { useSelector, useDispatch } from '../react-redux';
+import Label from './Label';
+import { getActiveAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getSession } from '../directSelectors';
+import { dismissCompatNotice } from '../session/sessionActions';
+import ZulipTextButton from './ZulipTextButton';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+const styles = createStyleSheet({
+  wrapper: {
+    backgroundColor: HALF_COLOR,
+    padding: 8,
+  },
+  textRow: {
+    padding: 8,
+    flexDirection: 'row',
+  },
+  buttonsRow: {
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+});
```

Comment:
From the spec:
https://material.io/components/banners#specs

I read `leftPadding: 16` for the wrapper (while 8 at right and bottom, and the top is specced in different terms that I'm not sure we can get in RN). Then no additional padding for `textRow`, except 12px vertical space between it and `buttonsRow`.

Speculating, I think the idea in giving the text more left padding than right padding may be that because it's justified left, there's naturally some space between the actual text and the right edge of its box -- and so the text will look more centered if the box is actually somewhat right of center.

Reply:
> _Spec_-ulating

(lol)

> the top is specced in different terms that I'm not sure we can get in RN

Looks like maybe we can get that by applying this to the text:

```js
  text: {
    marginTop: 16,
    lineHeight: 20,
  },
```

I'll give it a go in my next revision.